platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/StandaloneEventValidator.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.validation;
+
+import static com.swirlds.common.metrics.Metrics.PLATFORM_CATEGORY;
+import static com.swirlds.platform.event.validation.EventValidationChecks.areParentsValid;
+import static com.swirlds.platform.event.validation.EventValidationChecks.isGenerationValid;
+import static com.swirlds.platform.event.validation.EventValidationChecks.isSignatureValid;
+import static com.swirlds.platform.event.validation.EventValidationChecks.isValidTimeCreated;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.metrics.LongAccumulator;
+import com.swirlds.common.system.SoftwareVersion;
+import com.swirlds.common.system.address.AddressBook;
+import com.swirlds.common.utility.throttle.RateLimitedLogger;
+import com.swirlds.platform.crypto.SignatureVerifier;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.gossip.IntakeEventCounter;
+import com.swirlds.platform.internal.EventImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.Consumer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Validates events.
+ * <p>
+ * This validator is "standalone", because it is separate from the legacy intake monolith.
+ */
+public class StandaloneEventValidator {
```

Comment:
Make a mental note to come and rename this to just `Event Validator` once it is the only implementation.

Reply:
Made